### PR TITLE
API-4516: Remove ForwardedHeaders from outgoing notifications

### PR DESCRIPTION
**Pull request by cjrowe**

### CODE REVIEW

File: app/uk/gov/hmrc/pushpullnotificationsgateway/models/OutboundNotification.scala [hunk 1]
```diff
@@ -16,5 +16,4 @@
 
 package uk.gov.hmrc.pushpullnotificationsgateway.models
 
-case class ForwardedHeader(key: String, value: String)
-case class OutboundNotification(destinationUrl: String, forwardedHeaders: List[ForwardedHeader], payload: String)
+case class OutboundNotification(destinationUrl: String, payload: String)
```
Comment (lburgos) [inline app/uk/gov/hmrc/pushpullnotificationsgateway/models/OutboundNotification.scala]: shouldn't this be "case class OutboundNotification(destinationUrl: String, payload: NotificationResponse)"? Then the connector would automatically add the json content type

File: app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala [hunk 2]
```diff
@@ -77,7 +77,7 @@ class OutboundProxyConnector @Inject()(appConfig: AppConfig,
     def failedRequestLogMessage(statusCode: Int) = s"Attempted request to ${notification.destinationUrl} responded with HTTP response code $statusCode"
     implicit val hc: HeaderCarrier =  HeaderCarrier()
     validateDestinationUrl(notification.destinationUrl) flatMap { validatedDestinationUrl =>
-      httpClient.POST[String, HttpResponse](validatedDestinationUrl, notification.payload, notification.forwardedHeaders.map(fh => (fh.key, fh.value)))
+      httpClient.POST[String, HttpResponse](validatedDestinationUrl, notification.payload)
```
Comment (lburgos) [inline app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala]: I think the request type being a string, it won't add the json content type automatically, unless you've tested this and it works somehow.

Comment (lburgos) [reply app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala]: Maybe it's easier to just add the content type explicitly as we won't really use the notification model for anything.

Comment (cjrowe) [reply app/uk/gov/hmrc/pushpullnotificationsgateway/connectors/OutboundProxyConnector.scala]: That's true - I'll fix that

File: app/uk/gov/hmrc/pushpullnotificationsgateway/models/OutboundNotification.scala (hunk 1 above)
Comment (cjrowe) [reply app/uk/gov/hmrc/pushpullnotificationsgateway/models/OutboundNotification.scala]: Heh, there wasn't a `NotificationResponse` type in Gateway...

**Review by lburgos — APPROVED** [review]

**Pull request merged**